src/Components/Applications/Form/index.js

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from 'react';
+import styles from './form.module.css';
+function Form() {
+  const [positionId, setPositionId] = useState('');
+  const [postulantId, setPostulantId] = useState('');
+  const [date, setDate] = useState('');
+  const [result, setResult] = useState('');
+  const [positions, setPositions] = useState([]);
+  const [postulants, setPostulants] = useState([]);
+  const [interviews, setInterviews] = useState([]);
+
+  const params = new URLSearchParams(window.location.search);
+  const applicationId = params.get('_id');
+  let fetchMethod = 'POST';
+  let disableButton = false;
+
+  const onLoading = (data) => {
+    setPostulantId(data.data[0].postulants ? data.data[0].postulants._id : '');
+    setPositionId(data.data[0].positions ? data.data[0].positions._id : '');
+    setDate(data.data[0].interview ? data.data[0].interview.date : '');
+    setResult(data.data[0].result);
+  };
+  const onChangePositionId = (event) => {
+    setPositionId(event.target.value);
+  };
+  const onChangePostulantId = (event) => {
+    setPostulantId(event.target.value);
+  };
+  const onChangeDate = (event) => {
+    setDate(event.target.value);
+  };
+  const onChangeResult = (event) => {
+    setResult(event.target.value);
+  };
+
+  if (applicationId) {
+    fetchMethod = 'PUT';
+    disableButton = true;
+  }
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    const params = new URLSearchParams(window.location.search);
+    const applicationId = params.get('_id');
+    const options = {
+      method: fetchMethod,
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        positions: positionId,
+        postulants: postulantId,
+        interview: date,
+        result: result
+      })
+    };
+    const url = applicationId
+      ? `${process.env.REACT_APP_API}/applications/${applicationId}`
+      : `${process.env.REACT_APP_API}/applications/`;
+    fetch(url, options)
+      .then((response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          return response.json().then(({ message }) => {
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      })
+      .then(() => {
+        window.location.href = '/applications';
+      })
+      .catch((error) => {
+        console.log('err', error);
+      });
+  };
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API}/positions`)
+      .then((response) => response.json())
+      .then((res) => {
+        setPositions(res.data);
+      });
+    fetch(`${process.env.REACT_APP_API}/postulants`)
+      .then((response) => response.json())
+      .then((res) => {
+        setPostulants(res.data);
+      });
+    fetch(`${process.env.REACT_APP_API}/interviews`)
+      .then((response) => response.json())
+      .then((res) => {
+        setInterviews(res.data);
+      });
```

Comment:
also you should handle and show the errors
```suggestion
    fetch(`${process.env.REACT_APP_API}/interviews`)
      .then((response) => {
        if (response.status !== 200) {
          return response.json().then(({ message }) => {
            throw new Error(message);
          });
        }
        return response.json();
      })
      .then((response) => {
        setInterviews(response.data);
      })
      .catch((error) => {
        // Do something with the error
      });
```